### PR TITLE
executor: support "show columns from" for view

### PR DESCRIPTION
**Pull request by AndrewDi**
### What problem does this PR solve? 
Modify `DESCRIBE TABLE` to describe view columns

### What is changed and how it works?
ref proposal [Proposal: Implement View](https://github.com/pingcap/tidb/blob/master/docs/design/2018-10-24-view-support.md)

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change
Export function `func (b *PlanBuilder) BuildDataSourceFromView(dbName model.CIStr, tableInfo *model.TableInfo)` 

Side effects
none

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/8863)

### CODE REVIEW

**Comment by AndrewDi** [comment]
@XuHuaiyu @winkyao PTAL

**Comment by AndrewDi** [comment]
@XuHuaiyu Show_test do not contain any `DESCRIBE TBALE` test, so I didn't add test either.

**Comment by XuHuaiyu** [comment]
We'd better add some test cases for it.

**Comment by AndrewDi** [comment]
> We'd better add some test cases for it.

Done

**Comment by AndrewDi** [comment]
@XuHuaiyu PTAL

File: executor/show.go [hunk 1]
```diff
@@ -329,6 +330,20 @@ func (e *ShowExec) fetchShowColumns() error {
...
+		planBuilder := plannercore.NewPlanBuilder(e.ctx, e.is)
+		viewLogicalPlan, err := planBuilder.BuildDataSourceFromView(e.DBName, tb.Meta())
+		if err != nil {
+			return err
+		}
+		viewSchema := viewLogicalPlan.Schema()
+		for _, col := range cols {
+			viewColumn := viewSchema.FindColumnByName(col.Name.L)
+			if viewColumn != nil {
+				col.FieldType = *viewColumn.GetType()
```
Comment (zz-jason) [inline executor/show.go]: We can't get the column type from the columns returned by `tb.Cols()`?

Comment (AndrewDi) [reply executor/show.go]: Yes, view's column tableinfo only store column name, so we need to extract sql's schema.column to get column type.

Comment (zz-jason) [reply executor/show.go]: Can we fill the column type info when this table info is generated?

Comment (AndrewDi) [reply executor/show.go]: I don't think so, because undertable's column type and column length can be change.

Comment (zz-jason) [reply executor/show.go]:
```sql
MySQL(root@localhost:test) > create table t(a bigint, b bigint);
Query OK, 0 rows affected (0.04 sec)

MySQL(root@localhost:test) > create view v as select * from t;
Query OK, 0 rows affected (0.07 sec)

MySQL(root@localhost:test) > show columns from v;
+-------+------------+------+-----+---------+-------+
| Field | Type       | Null | Key | Default | Extra |
+-------+------------+------+-----+---------+-------+
| a     | bigint(20) | YES  |     | NULL    |       |
| b     | bigint(20) | YES  |     | NULL    |       |
+-------+------------+------+-----+---------+-------+
2 rows in set (0.01 sec)

MySQL(root@localhost:test) > drop table t;
Query OK, 0 rows affected (0.03 sec)

MySQL(root@localhost:test) > create table t(a double, b double, c double);
Query OK, 0 rows affected (0.07 sec)

MySQL(root@localhost:test) > show columns from v;
+-------+--------+------+-----+---------+-------+
| Field | Type   | Null | Key | Default | Extra |
+-------+--------+------+-----+---------+-------+
| a     | double | YES  |     | NULL    |       |
| b     | double | YES  |     | NULL    |       |
+-------+--------+------+-----+---------+-------+
2 rows in set (0.01 sec)
```

Got it. Confirmed in MySQL.

Comment (zz-jason) [reply executor/show.go]: @AndrewDi Could you add a comment to specify why we need to build logical plan on the view to get the column types?

Comment (AndrewDi) [reply executor/show.go]: Done

**Pull request merged**